plugins/matplotlib/src/DashboardPlugin.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useCallback, useEffect } from "react";
+import shortid from "shortid";
+import { LayoutUtils, useListener } from "@deephaven/dashboard";
+import MatPlotLibPanel from "./MatPlotLibPanel";
+
+export const MatPlotLibVariableType = "matplotlib.figure.Figure";
+
+export const DashboardPlugin = ({ id, layout, registerComponent }) => {
+  const handlePanelOpen = useCallback(
+    ({ dragEvent, fetch, panelId = shortid.generate(), widget }) => {
+      const { name, type } = widget;
+      if (type !== MatPlotLibVariableType) {
+        return;
+      }
+      const metadata = { name, type };
+      const config = {
+        type: "react-component",
+        component: MatPlotLibPanel.COMPONENT,
+        props: {
+          localDashboardId: id,
+          id: panelId,
+          metadata,
+          fetch,
+        },
+        title: name,
+        id: panelId,
+      };
+
+      const { root } = layout;
+      LayoutUtils.openComponent({ root, config, dragEvent });
+    },
+    [id, layout],
+  );
```

Comment:
This is extremely similar to the Object panel plugin code. Is this some logic we should abstract out to a hook?

Reply:
We could. Notice the type check for object panel is an exclusion list rather than an inclusion list.